src/tools/rados/rados.cc

```diff
@@ -2144,6 +2137,13 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
     }
   }
 
+  i = opts.find("pgid");
+  boost::optional<pg_t> pgid(i != opts.end(), pg_t());
+  if (pgid && (!pgid->parse(i->second.c_str()) || (pool_name && rados.pool_lookup(pool_name) != pgid->pool()))) {
```

Comment:
@vumrao could you note in the commit message that the reason of crash is that we cannot use the `rados` object before connecting it with cluster using `rados.init_with_context()`?

Reply:
@tchaikov thanks Kefu. I have made the changes in the commit message.